src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -3149,9 +3160,29 @@ ProduceBoundNode:
                             If container IsNot Nothing AndAlso container.Name IsNot Nothing Then
                                 callerInfoValue = ConstantValue.Create(container.Name)
                             End If
-                        Else
-                            Debug.Assert(isCallerFilePath)
+                        ElseIf isCallerFilePath Then
                             callerInfoValue = ConstantValue.Create(callerInfoOpt.SyntaxTree.GetDisplayPath(callerInfoOpt.Span, Me.Compilation.Options.SourceReferenceResolver))
+                        Else
+                            Debug.Assert(callerArgumentExpressionParameterIndex > -1 OrElse reducedExtensionReceiverOpt IsNot Nothing)
```

Comment:
>reducedExtensionReceiverOpt IsNot Nothing [](http://example.com/codeflow?start=92&length=41)

AndAlso?  ```callerArgumentExpressionParameterIndex > -2```?

Reply:
@AlekseyTs OrElse is correct here.
`reducedExtensionReceiverOpt` is okay to be null here.

The idea of this assert is:

- Do we have an index >= 0? If yes, we don't care about `reducedExtensionReceiverOpt`
- Otherwise, if we have `-1`, then this must be the receiver of an extension method, in which case `reducedExtensionReceiverOpt` must be non-null value.

Reply:
>  OrElse is correct here.

The suggestion is to strengthen the second operand of OrElse, not to replace OrElse

Reply:
I misinterpreted the original comment. Sorry, and thanks for clarifying. I updated the assert.